React/Base/RCTUtils.m

```diff
@@ -337,6 +337,42 @@ BOOL RCTRunningInTestEnvironment(void)
   return isTestEnvironment;
 }
 
+BOOL RCTRunningInAppExtension(void)
+{
+  return [[[[NSBundle mainBundle] bundlePath] pathExtension] isEqualToString:@"appex"];
+}
+
+id RCTSharedApplication(void)
+{
+  if (RCTRunningInAppExtension()) {
+    return nil;
+  }
+  
+  return [[UIApplication class] performSelector:@selector(sharedApplication)];
```

Comment:
The UIAlertView slips through because it's using `init` instead of the blacklisted `initWithTitle:message:etc...` selector.  As far as I can tell, the `NS_EXTENSION_UNAVAILABLE_IOS` compiler flag just applies to method selectors, so we can work around flagged initializers by just calling `init` (or `new`), which is too generic to be flagged.

I think (but could be mistaken) that performSelector and ARC are only problematic if the selector isn't known at compile time (i.e. you're using `NSSelectorFromString` or something).  Xcode is apparently smart enough to determine the return type and apply the correct ARC rules when you use `@selector(selectorName)`, but not smart enough to apply the extension API blacklisting.